src/state/reducers/codeSystem.ts

```diff
@@ -0,0 +1,52 @@
+import { CodeSystem } from "../../resources/ts/proto/r4/core/resources/code_system_pb";
+import {
+  GET_CODESYSTEMDATATYPE_SUCCESS,
+  GET_CODESYSTEMDATATYPE_FAILURE,
+  GetCodeSystemDataTypeSuccessAction,
+  GetCodeSystemDataTypeFailureAction
+} from "../actions/codeSystemActions";
+
+import { String } from "../../resources/ts/proto/r4/core/datatypes_pb";
+
+export type DataFetched = {
+  id: String.AsObject;
+};
+
+export type CodeSystemState = {
+  datatypes: CodeSystem.AsObject | null;
+  loading: boolean;
+  error: Error | null;
+};
+
+const initialState: CodeSystemState = {
+  datatypes: null,
+  loading: false,
```

Comment:
This 'loading' attribute is never set to 'true'.

If you want to use it properly, it should be set to true from an action dispatched by the thunkAction you created. For instance, by creating an action like 'getCodeSystemDataTypePending' and dispatching it in the first line of your thunkAction would allow this reducer to set the 'loading' state attribute to true as soon as the API request starts.